CMakeLists.txt

```diff
@@ -239,8 +239,11 @@ if(USE_MKLDNN)
   if(NOT MSVC)
     set(ARCH_OPT_FLAGS "-mtune=generic")
   endif()
-  set(WITH_TEST OFF)
-  set(WITH_EXAMPLE OFF)
+  set(WITH_TEST OFF CACHE INTERNAL "" FORCE)
+  set(WITH_EXAMPLE OFF CACHE INTERNAL "" FORCE)
+  set(ARCH_OPT_FLAGS "" CACHE INTERNAL "" FORCE)
+
+  include_directories(3rdparty/mkldnn)
```

Comment:
Just curious, is there a reason relevant headers aren't in mkldnn/include?

Reply:
Some headers in `src/common` are used for MKL-DNN tests. Previously, `set(WITH_TEST OFF)` doesn't work for MKL-DNN build. Those MKL-DNN tests will still be compiled. I change the setting for `WITH_TEST` and `WITH_EXAMPLE` here, so maybe this line can be removed. I will try.